arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Either.kt

```diff
@@ -2292,19 +2292,43 @@ public operator fun <A : Comparable<A>, B : Comparable<B>> Either<A, B>.compareT
     { b1 -> other.fold({ 1 }, { b2 -> b1.compareTo(b2) }) }
   )
 
+/**
+ * Combine two [Either] values.
+ * If both are [Right] then combine both [B] values using [combineRight] or if both are [Left] then combine both [A] values using [combineLeft],
+ * otherwise it returns the `this` or fallbacks to [other] in case `this` is [Left].
+ */
+public fun <A, B> Either<A, B>.combine(other: Either<A, B>, combineLeft: (A, A) -> A, combineRight: (B, B) -> B): Either<A, B> =
+  when (val one = this) {
+    is Left -> when (other) {
+      is Left -> Left(combineLeft(one.value, other.value))
+      is Right -> one
+    }
+
+    is Right -> when (other) {
+      is Left -> other
+      is Right -> Right(combineRight(one.value, other.value))
+    }
+  }
+
 @Deprecated(
-  RedundantAPI + "Prefer zipOrAccumulate",
-  ReplaceWith("Either.zipOrAccumulate({ a, bb -> SGA.run { a.combine(bb) }  }, this, b) { a, bb -> SGB.run { a.combine(bb) } }")
+  SemigroupDeprecation,
+  ReplaceWith(
+    "Either.zipOrAccumulate(SGA::combine, this, b, SGB::combine)",
+    "arrow.typeclasses.combine"
+  )
```

Comment:
**thought:** 

<sub>This comment follows the [**conventionalcomments.org**](https://conventionalcomments.org/) standard</sub>
```suggestion
  SemigroupDeprecation,
  ReplaceWith("this.combine(b, {(a:A, a:A) -> A}, {(b:B, b:B) -> B})")
```

Maybe changing the deprecation expression, we can help a bit more the user with the migrations. It does not compile, but the current expression does not either.

Reply:
I don't fully understand: the `ReplaceWith` expression is a copy of the implementation. I agree that it may not compile in every case, but I think this gets the developer closer to the expected changes than otherwise.